src/main/ruby/core/io.rb

```diff
@@ -2667,7 +2673,7 @@ def close
       if fd >= 0
         # Need to set even if the instance is frozen
         Truffle.invoke_primitive :object_ivar_set, self, :@descriptor, -1
-        if fd >= 3
+        if fd >= 3 && false != @autoclose
```

Comment:
Is there any value in comparing directly to `false`? Using the `autoclose?` helper strikes me as being clearer and would draw my eye up to the nice comment you added there.

Reply:
I saw a `nil` value of `@autoclose` on some IO objects, but I tracked that down and fixed it so have changed this to simply call `autoclose?`